pkg/pipeline/CiCdPipelineOrchestrator.go

```diff
@@ -1430,6 +1474,15 @@ func (impl CiCdPipelineOrchestratorImpl) GetCdPipelinesForEnv(envId int, request
 			postStage.TriggerType = dbPipeline.PostTriggerType
 			pipeline.PostStage = postStage
 		}
+
+		preDeployStage, postDeployStage, err := impl.pipelineStageService.GetCdPipelineStageDataDeepCopy(dbPipeline.Id)
```

Comment:
we need to optimize this, we should fetch for multiple pipelines in a single request

Reply:
optimized